logstash-core-plugin-api/logstash-core-plugin-api.gemspec

```diff
@@ -0,0 +1,27 @@
+# -*- encoding: utf-8 -*-
+lib = File.expand_path('../lib', __FILE__)
+$LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
+require "logstash-core-plugin-api/version"
+# This is completely valid, when we do a `gem build` the gemspec is 
+# converted to a text file without any ruby code.
+require_relative "../logstash-core/lib/logstash-core/version"
+
+Gem::Specification.new do |gem|
+  gem.authors       = ["Elasticsearch"]
+  gem.email         = ["info@elasticsearch.com"]
+  gem.description   = %q{Logstash plugin api contract}
+  gem.summary       = %q{Define the plugin contract that the plugin need to follow.}
+  gem.homepage      = "http://www.elastic.co/guide/en/logstash/current/index.html"
+  gem.license       = "Apache License (2.0)"
+
+  gem.files         = Dir.glob(["logstash-core-event.gemspec", "lib/**/*.rb", "spec/**/*.rb"])
+  gem.test_files    = gem.files.grep(%r{^(test|spec|features)/})
+  gem.name          = "logstash-core-plugin-api"
+  gem.require_paths = ["lib"]
+  gem.version       = LOGSTASH_CORE_PLUGIN_API
+
+  gem.add_runtime_dependency "logstash-core", "#{LOGSTASH_CORE_VERSION}"
```

Comment:
In this (proposition)[https://github.com/elastic/logstash/issues/4829#issuecomment-199925565] I was proposing the following constraints, I agree it might not be clear with the `satisfied_by?`

```
 gem.add_runtime_dependency "logstash-core", ">= 2.0.0" ,"<= #{LOGSTASH_CORE_VERSION}"
```

But after discussing with you yesterday, I though to use a strict dependency on a specific `logtash-core` and not use a range. But this constraints cannot work for the first version of `logstash-core-plugin-api` (1.0.0) IT **needs** to be  a range like this.

```
 gem.add_runtime_dependency "logstash-core", ">= 2.0.0" ,"<= 2.3.0"
```

If we don't do this, plugins wont be able to be installed in a pre Logstash 2.3 installation.